cluster-up/cluster/k3d/common.sh

```diff
@@ -2,312 +2,173 @@
-    fi
-}
+function _create_cluser() {
+    echo "STEP: Create cluster"
 
-function _prepare_kind_config() {
-    _add_workers
-    _add_kubeadm_config_patches
+    id1=${BASE_PATH}/$KUBEVIRT_PROVIDER/machine-id-1
+    id2=${BASE_PATH}/$KUBEVIRT_PROVIDER/machine-id-2
+    id3=${BASE_PATH}/$KUBEVIRT_PROVIDER/machine-id-3
+    printf '%.0s1' {1..32} > ${id1}
+    printf '%.0s2' {1..32} > ${id2}
+    printf '%.0s3' {1..32} > ${id3}
 
-    echo "Final KIND config:"
-    cat ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/kind.yaml
-}
+    NETWORK=bridge
+    if [ $CRI_BIN == podman ]; then
+        NETWORK=podman
+        # TODO need to update according K3D_TAG or fix with a nicer solution
+        podman pull docker.io/rancher/k3s:v1.25.6-k3s1
+    fi
 
-function kind_up() {
-    _fetch_kind
-    _prepare_kind_config
-    setup_kind
+    k3d registry create --default-network $NETWORK $REGISTRY_NAME --port $REGISTRY_HOST:$HOST_PORT
+    ${CRI_BIN} rename k3d-$REGISTRY_NAME $REGISTRY_NAME
+
+    k3d cluster create $CLUSTER_NAME --registry-use $REGISTRY_NAME \
+        --api-port $KUBERNETES_SERVICE_HOST:$KUBERNETES_SERVICE_PORT \
+        --servers=$KUBEVIRT_NUM_SERVERS \
+        --agents=$KUBEVIRT_NUM_AGENTS \
+        --k3s-arg "--disable=traefik@server:0" \
+        --no-lb \
+        --k3s-arg "--flannel-backend=none@server:*" \
+        --k3s-arg "--kubelet-arg=cpu-manager-policy=static@agent:*" \
+        --k3s-arg "--kubelet-arg=kube-reserved=cpu=500m@agent:*" \
+        --k3s-arg "--kubelet-arg=system-reserved=cpu=500m@agent:*" \
+        --volume "$(pwd)/cluster-up/cluster/k3d/manifests/calico.yaml:/var/lib/rancher/k3s/server/manifests/calico.yaml@server:0" \
+        -v /dev/vfio:/dev/vfio@agent:* \
+        -v /lib/modules:/lib/modules@agent:* \
+        -v ${id1}:/etc/machine-id@server:0 \
+        -v ${id2}:/etc/machine-id@agent:0 \
+        -v ${id3}:/etc/machine-id@agent:1
+}
+
+function k3d_up() {
+    _install_k3d
+    _create_cluser
+    _extract_kubeconfig
+    _prepare_nodes
+    _install_cni_plugins
+    _prepare_provider_config
+    _label_agents
 }
 
 function _kubectl() {
-    ${KUBECTL} "$@"
+    export KUBECONFIG=${BASE_PATH}/$KUBEVIRT_PROVIDER/.kubeconfig
+    kubectl --kubeconfig=$KUBECONFIG "$@"
 }
 
 function down() {
-    _fetch_kind
-    if [ -z "$($KIND get clusters | grep ${CLUSTER_NAME})" ]; then
-        return
-    fi
-    # On CI, avoid failing an entire test run just because of a deletion error
-    $KIND delete cluster --name=${CLUSTER_NAME} || [ "$CI" = "true" ]
-    ${CRI_BIN} rm -f $REGISTRY_NAME >> /dev/null
-    rm -f ${KUBEVIRTCI_CONFIG_PATH}/$KUBEVIRT_PROVIDER/kind.yaml
+    set +e
+    trap "set -e" RETURN
+
+    for agent_node in $(_get_agent_nodes); do
+        if ip netns exec $agent_node ip -d a | grep "vf 0" -B 2 > /dev/null; then
+            iface=$(ip netns exec $agent_node ip -d a | grep "vf 0" -B 2 | grep "UP" | awk -F": " '{print $2}')
```

Comment:
Can you use full names for `-d`, `a` etc... It is not clear what is it or what this part does.
All the magic numbers is also unclear to me.